db/src/lib.rs

```diff
@@ -35,6 +32,34 @@ pub async fn create_user(pool: &PgPool, user_name: &str) -> color_eyre::Result<U
     Ok(result.user_id)
 }
 
+pub async fn add_url(pool: &PgPool, url: &str, allow_existing: &bool) -> color_eyre::Result<String> {
+    let exist_record = sqlx::query_as!(
+        ExistRecord,
+        "SELECT EXISTS(SELECT 1 FROM Page WHERE url = $1) as exists",
+        url
+    )
+    .fetch_one(pool)
+    .await?;
+
+    // exist_record might be true or false
+    if let Some(true) = exist_record.exists {
+        if *allow_existing {
+            return Ok(format!("URL already exists and re-adding is allowed: {url}"));
+        } else {
+            return Err(color_eyre::eyre::eyre!("URL already exists and re-adding is not allowed."));
+        }
+    } 
+    
+    let result = sqlx::query!(
+        "INSERT INTO Page (url) VALUES ($1) RETURNING url",
+        url
+    )
+    .fetch_one(pool)
+    .await?;
+
+    Ok(result.url)
```

Comment:
Nit: This is obviously perfect for this use case!

But forward thinking a bit, I wonder if these 'create' method should/could return the full object on create?

Like we could create a `Page` struct with fields that match the table and return a 'full' Page struct here! That way the consumer could get either the url or the ids depending on what they need